services/jetbrains/jetbrains-base.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseXmlService = require('../base-xml')
+const { NotFound } = require('../errors')
+
+module.exports = class JetbrainsBase extends BaseXmlService {
+  static buildUrl(base) {
+    return {
+      base,
+      format: '(.+)',
+      capture: ['pluginId'],
+    }
+  }
+
+  async fetchPackageData({ pluginId, schema }) {
+    const parserOptions = {
+      parseNodeValue: false,
+      ignoreAttributes: false,
+    }
+    const pluginData = await this._requestXml({
+      schema: Joi.object({
+        'plugin-repository': Joi.any().required(),
+      }),
+      url: `https://plugins.jetbrains.com/plugins/list?pluginId=${pluginId}`,
+      parserOptions,
+    })
+    if (!pluginData['plugin-repository']) {
```

Comment:
Why is this being checked manually? Is the response code returned by JetBrains unhelpful? If that's the case, I would suggest using an approach similar to the _aur_ service, i.e. override `_validate`, check the data for not found, then call the parent's `_validate` method. This will make things clearer and will avoid having different Joi schemas being used in different parts of the implementation.